### PR TITLE
Add: ANGSD_DOCOUNTS

### PR DESCRIPTION
**Pull request by jfy133**
First module for #841 

Note ANGSD is extremely messy to use (it doesnt follow 'standard' unix CLI structure), and appears to have lots of undocumented input parameters/requirements etc, so this module and others (if added) will likely require lots of iterations until they are able to work with all inputs/outputs...

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/angsd/docounts/main.nf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    tuple val(meta), path(bam), path(bai), path(minqfile)
+
+    output:
+    tuple val(meta), path("*.depthSample"), optional: true, emit: depthsample
+    tuple val(meta), path("*.depthGlobal"), optional: true, emit: depthglobal
+    tuple val(meta), path("*.qs")         , optional: true, emit: qs
+    tuple val(meta), path("*.pos.gz")     , optional: true, emit: pos
+    tuple val(meta), path("*.counts.gz")  , optional: true, emit: counts
+    tuple val(meta), path("*.icnts.gz")   , optional: true, emit: icounts
+
```
Comment (jfy133) [inline modules/angsd/docounts/main.nf:20]:
```suggestion
```

File: modules/angsd/docounts/meta.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      pattern: "versions.yml"
+  - depthsample:
+      type: file
+      description: Distribution of sequencing depths
+      pattern: "*.depthSample"
+  - depthglobal:
+      type: file
+      description: Distribution of sequencing depths
+      pattern: "*.depthGlobal"
+  - qs:
```
Comment (jfy133) [inline modules/angsd/docounts/meta.yml:55]:
```suggestion
  - depth_sample:
      type: file
      description: Distribution of sequencing depths
      pattern: "*.depthSample"
  - depth_global:
      type: file
      description: Distribution of sequencing depths
      pattern: "*.depthGlobal"
  - qs:
```

Update channel names

File: modules/angsd/docounts/meta.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      description: File with individual quality score thresholds
+      pattern: "*"
+
+output:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  #
```
Comment (SPPearce) [inline modules/angsd/docounts/meta.yml:42]: Is there a reason for this blank line?

**Review by SPPearce — APPROVED** [review]
A couple of minor suggestions, but looks good to me.

File: modules/angsd/docounts/meta.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
+name: "angsd_docounts"
+description: Sometimes we want or need the frequency of the different bases. This is what -doCounts does.
+keywords:
+  - angsd
+  - population genetics
+  - allele counts
+  - doCounts
```
[the remark this answers is not in the document]
Comment (jfy133) [reply modules/angsd/docounts/meta.yml:7]: I'm not enitrely sure so I didn't add that tag

File: modules/angsd/docounts/meta.yml (hunk 3 above)
Comment (jfy133) [reply modules/angsd/docounts/meta.yml:42]: Nope, was from template - forgot to remove it!

Comment (jfy133) [inline modules/angsd/docounts/meta.yml:42]:
```suggestion
```

File: modules/angsd/docounts/meta.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+name: "angsd_docounts"
+description: Sometimes we want or need the frequency of the different bases. This is what -doCounts does.
```
Comment (jfy133) [inline modules/angsd/docounts/meta.yml:2]:
```suggestion
description: Calculates base frequency statistics across reference positions from BAM and VCF files.
```

File: modules/angsd/docounts/meta.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+name: "angsd_docounts"
+description: Calculates base frequency statistics across reference positions from BAM and VCF files.
```
Comment (jfy133) [inline modules/angsd/docounts/meta.yml:2]:
```suggestion
description: Calculates base frequency statistics across reference positions from BAM.
```

**Pull request merged**